torch/onnx/symbolic_opset9.py

```diff
@@ -3156,3 +3156,36 @@ def index_add(g, self, dim, index, other):
         index = sym_help._unsqueeze_helper(g, index, [sym_help._get_tensor_rank(index)])
 
     return scatter_add(g, self, dim, expand_as(g, index, other), other)
+
+
+@parse_args('v', 'is', 'is')
+def roll(g, self, shifts, dims):
+    assert len(shifts) == len(dims)
+
+    dim_size_list = [sym_help._get_tensor_dim_size(self, dims[i]) for i in range(len(dims))]
+
+    result = self
+    for i in range(len(shifts)):
+        if dim_size_list[i] is None:
+            raise NotImplementedError("ONNX export does NOT support exporting 'roll()' function with " +
+                                      "unknown dimension size.")
+
+        if shifts[i] < 0:
+            shifts[i] = shifts[i] + dim_size_list[i]
+
+        shapes = []
```

Comment:
I think it could be possible to convert without knowing `dim_size_list[i]`. Consider construct the two slices like
`slice[-shift, maxsize] + slice[0, -shift]`. It applies both for negative and positive `shift`.

Reply:
This is great. It works as expected.